### ISSUE TITLE
Number of trees (RandomForest + GBT) not in config space

### ISSUE THREAD

**Issue by TaXxER** [automl/auto-sklearn#236]
I noticed that the number of trees parameter of RandomForest and Gradient Boosted Trees is fixes to 100 trees, while with manual tuning I generally am able to obtain much better performance with higher numbers of trees on the data set that I am currently analyzing. The time-out that I set per model trained allows for many more than 100 trees to be trained before the time-out will kick-in. Is it possible to extend the config space such that it explores higher number of trees? Or at least fix this parameter to a higher value than 100? I was not able to find anything on this in the documentation.

**Comment by mfeurer** [comment]
I do see your point, but I don't know of an easy solution for this problem. Would you be willing to invest some time in this problem? Otherwise, you could change the value in your installation and I can point you to the files you have to change.

**Comment by TaXxER** [comment]
Unfortunately I can not make the time available, at least at this point in time, to solve this problem properly. Could you point me to the files that I have to change?

**Comment by mfeurer** [comment]
Files are in the module [autosklearn.pipeline.components.classification](https://github.com/automl/auto-sklearn/tree/master/autosklearn/pipeline/components/classification). Each file there represents one classifier, and in the methods [get_hyperparameter_search_space](https://github.com/automl/auto-sklearn/blob/master/autosklearn/pipeline/components/classification/random_forest.py#L124) you can change the bounds and values of the hyperparameters.

**Comment by mfeurer** [comment]
I'm closing this now as the issue is quite old by now, but I'm leaving a few comments that are hopefully helpful:

1. We now have an example which demonstrates how to achieve this: https://automl.github.io/auto-sklearn/master/examples/80_extending/example_restrict_number_of_hyperparameters.html
2. We increased the number to 512 and use early stopping where possible
3. Auto-sklearn 2 can use successive halving which gradually increases the number of trees, making a tradeoff between spending a lot of time on all configurations and cutting of potentially good configurations early on.
